python/astshim/test.py

```diff
@@ -11,10 +11,21 @@
 
 
 class ObjectTestCase(unittest.TestCase):
-
     """Base class for unit tests of objects
     """
 
+    def assertObjectsIdentical(self, obj1, obj2):
+        """Assert that two astshim objects are identical.
+
+        Identical means the objects are of the same class
+        and all properties are identical
+        (including whether set or defaulted).
+        """
+        self.assertEqual(obj1.className, obj2.className)
```

Comment:
Things can get a bit weird for astshim classes with no direct AST equivalent: `FrameDict` is a wrapper around `FrameSet` and `SeriesMap` and `ParallelMap` are both basically `CmpMap`s.

However, I decided you were right and I made the test picker while adding support for the type being something other than expected in `checkPersistence`.

This also prompted me to improve pickle support for `FrameDict` so it round-trips correctly instead of coming back as a `FrameSet`.